### PR TITLE
Use `top_level_semantic` in doc spec instead of `semantic`

### PR DESCRIPTION
**Pull request by makenowjust**
This PR adds `top_level_semantic` utility methods for compiler spec, and uses this in doc spec. `crystal docs` command [uses `top_level_semantic`](https://github.com/crystal-lang/crystal/blob/b3ebb1cde9c08eb15d3401775b7d5b8811335d7f/src/compiler/crystal/command/docs.cr#L139), so the spec follows it.

### CODE REVIEW

**Review by nobodywasishere — COMMENTED** [review]
Everything looks good to me - though I don't have write access so I cannot merge this PR. Does this generate the exact same docs as just 'semantic'? And how kuch faster is this?

**Comment by straight-shoota** [comment]
I'm not sure about the `wants_doc = true` default because its different from anywhere else.
I suppose it makes kind of sense when it's only used for doc generator specs.
However then I'd maybe consider extracing a specific helper method for that and maybe returns a `Doc::Generator` instance?

**Review by nobodywasishere — COMMENTED** [review]
I don't think this should've been removed. It's why CI is broken

**Review by nobodywasishere — CHANGES_REQUESTED** [review]

**Review by straight-shoota — APPROVED** [review]
Specs are green now 🚀

File: spec/spec_helper.cr
Comment (HertzDevil) [inline spec/spec_helper.cr:84]: I don't think `wants_doc` should have a different default value from `::semantic`, as it is entirely possible we want to use `top_level_semantic` in other places

File: spec/spec_helper.cr
Comment (HertzDevil) [inline spec/spec_helper.cr:82]: Does this really need to be a distinct type from `SemanticResult`?

**Review by HertzDevil — APPROVED** [review]

**Pull request merged**
